src/clj/ataru/odw/odw_service.clj

```diff
@@ -31,16 +31,52 @@
       nil)))
 
 (defn- suoritusvuosi-one-of
-  [application-key answers ids]
-  (if-let [values (some #(let [v (get-in answers [% :value])]
-                           (cond (string? v)
-                                 [v]
-                                 (some? v)
-                                 (map first v)))
-                        ids)]
-    (mapv (partial parse-year application-key) values)
-    (do (log/warn (str "No answers to questions " (clojure.string/join ", " ids)) "in hakemus" application-key)
-        [nil])))
+  ([application-key answers ids]
+   (suoritusvuosi-one-of application-key answers ids false))
+  ([application-key answers ids mute-logging]
+    (if-let [values (some #(let [v (get-in answers [% :value])]
+                            (cond (string? v)
+                              [v]
+                              (some? v)
+                              (map first v)))
+                          ids)]
+      (mapv (partial parse-year application-key) values)
+      (do (when-not mute-logging
+            (log/warn (str "No answers to questions " (clojure.string/join ", " ids)) "in hakemus" application-key))
+        [nil]))))
+
+(defn- kaksoistutkinto-suoritusvuosi
+  "Double degree (secondary level) completion year resolving. General idea as follows:
+
+   - If completion year is in future, it is unknown if it matches both matriculation and vocational completions or just
+     either - it's an estimation, not a hard fact. So in case the completion is some time in future, pick just that
+     directly.
+   - Otherwise we can assume completion year is in past, in which case we compare the completion years of both
+     matriculation and vocational parts of the degree and pick later as that matches the final completion year in
+     secondary level double degree. If only one of these is present, `nil` is returned to indicate the degree is not
+     fully complete yet."
+  [application-key answers]
+  (let [expected-completion-years [; 1. yhteishaun tunnisteet
+                                   :0a6ba6b1-616c-492b-a501-8b6656900ebd    ; 2017-nyt
+                                   :22df6790-588f-4c45-8238-3ecfccdf6d93_1  ; nyt
```

Comment:
Näiden "valmistun hakukautena" tapausten kanssa ei valitettavasti pääse ihan näin helpolla, vaan tarvitaan samankaltainen tarkastelu kuin esim. "pohjakoulutus_yo_kansainvalinen_suomessa" kanssa. Eli tarkastellaan kysymyksen `:22df6790-588f-4c45-8238-3ecfccdf6d93` vastauksen arvoa, ja jos se on `"1"` niin käytetään haulta löytyvää `:hakukausiVuosi` tietoa suoritusvuotena.

Reply:
Onko tuo arvo varmasti `"1"`, `pohjakoulutus_yo_kansainvalinen_suomessa` ja `pohjakoulutus_yo_ulkomainen` vertaavat arvoon `"0"` ja jos tämä, niin sitten käyttävät `:hakukausiVuosi` tietoa.